File: palindrom.js
//palindrome 意思是從左邊唸到右邊跟右邊唸到左邊都一樣

function palindrome(stings) {
  left = 0;
  right = stings.length - 1;
  while (left !== right) {
    if (stings[left] === stings[right]) {
      left++;
      right--;
    } else {
      return false;
    }
  }
  return true;
}

x = "amanaplanacanalpanama";
console.log(palindrome(x));

File: palindrom.py
def palindrome(strings):
    left = 0
    right = len(x)-1
    while(left < right):
        if (x[right] != x[left]):
            return False
        else:
            left += 1
            right -= 1
    return True


if __name__ == '__main__':
    x = "amanaplanacanalpanama"
    print(palindrome(x))

